h2o-docs/src/booklets/v2_2015/source/GLMBooklet.tex

```diff
@@ -571,6 +571,100 @@ \subsubsection{Tweedie Models}
+The $W_j$ terms are all positive. The following figure plots for $\mu = 0.5, p=1.5, \phi =1. y=0.1$.
+
+\begin{figure}[h]
+\centering
+\includegraphics[scale=0.25]{images/dispersion_param_fig1.png}
+\end{figure}
+
+\subparagraph{$\alpha (y, \phi) \text{ when } p > 2$}
+
+Here, you have,
+
+$$a(y, \phi, p) = \frac{1}{\pi y}V(y,\phi, p) \quad \text{Equation 6}$$
+
+and $V = \sum^{\infty}_{k=1} V_k$ where
+
+$$V_k = \frac{T(1+\alpha k)\phi^{k(\alpha - 1)}(p-1)^{\alpha k}}{T(1+k)(p-2)^ky^{\alpha k}}(-1)^k \sin (-k\pi \alpha) \quad \text{Equation 7}$$
+
+Note that $0 < \alpha < 1$ for $p > 2$. The $V_k$ terms are both positive and negative. This will limit the numerical accuracy that is obtained in summing it as shown in the following image. Again, if weights are applied to each row of the dataset, $equation 6$ becomes:
+
+$$V_k = \frac{T(1+\alpha k)\phi^{k(\alpha -1)}(p-1)^{\alpha k}}{T(1+k)w^{k(\alpha -1)}(p-2)^ky^{\alpha k}}(-1)^k \sin (-k\pi \alpha) \quad \text{Equation 8}$$
+
+\waterExampleInR
+\lstinputlisting[style=R]{GLM_Vignette_code_examples/glm_tweedie_dispersion_example.R}
+
+\waterExampleInPython
+\lstinputlisting[style=python]{GLM_Vignette_code_examples/glm_tweedie_dispersion_example.py}
+
+\paragraph{Tweedie Variance Power Maximum Likelihood} 
+
+Depending on $p$, $y$, and $\phi$, different methods are used for this log likelihood estimation. To start, let:
+
+$$\xi = \frac{\phi}{y^{2-p}}$$
+
+If $p=2$, then it will use the log likelihood of the Gamma distribution:
+
+$$\log (p) = \begin{cases} - \infty & y=0 \\ \frac{1}{\phi} \log (\frac{1}{\phi \mu}) - \log \text{Gamma} \frac{1}{\phi} + \log (y)(\frac{1}{\phi} -1) + (-\frac{1}{\phi \mu} y) & y>0 \\\end{cases}$$
+
+If $p=3$, then it will use the inverse Gaussian distribution:
+
+$$\log (p) = \begin{cases} - \infty & y=0 \\ \frac{1}{2} \Big(-log (\phi \mu) \log (2 \pi) -3 \log \big( \frac{y}{\mu} - \frac{(\frac{y}{\mu} -1)^2}{\phi \mu \frac{y}{\mu}} \Big) - \log (\mu) & y>0 \\\end{cases}$$
+
+If $p<2$ and $\xi \leq 0.01$, then it will use the Fourier inversion method.
+
+If $p>2$ and $\xi \geq 1$, then it will also use the Fourier inversion method.
+
+Everything else will use the Series method. However, if the Series method fails (output of \texttt{NaN}), then it will try the Fourier inversion method instead.
+
+If both the Series method and Fourier inversion method fail, or if the Fourier inversion method was chosen based on the $\xi$ criterium and it failed, it will then estimate the log likelihood using the Saddlepoint approximation.
+
+Here is the general usage for Tweedie variance power maximum likelihood:
+\begin{itemize}
+\item with \texttt{fix\_tweedie\_variance\_power = False} and \\ \texttt{fix\_dispersion\_parameter = True} as it will optimize the Tweedie variance power.
+\item with \texttt{fix\_tweedie\_variance\_power = False} and \\ \texttt{fix\_dispersion\_parameter = False} as it will optimize both Tweedie variance power and the dispersion parameter.
+\end{itemize}
    
```

Comment:
I would add the description about the optimization procedure:

When estimating just Tweedie variance power we use the golden section search and once we find some small region we switch to the Newton's method. If the Newton's method fails (steps out of bounds found by the golden section search) we use the golden section search until convergence. (just for you for the context: In theory Newton's method should be much faster but Tweedie likelihood has some numerical issues so it's not nicely smooth and this gets even more pronounced in 1st and 2nd derivatives which are used by Newton's method. Golden section search is using just the log likelihood (no derivatives) so it's more stable but has asymptotically slower rate of convergence)

When we optimize both Tweedie variance power and dispersion we use the Nelder-Mead method with constraints so that Tweedie variance power $p > 1+10^{-10}$ and dispersion $\phi > 10^{-10}$. (This was merged yesterday but I think we can add it in the documentation in this PR; again for context Nelder-Mead is a derivative free optimization.)